cognitedata/reveal
# feat: Toggle visibility of 360 image markers

#### Type of change
![Feat](https://img.shields.io/badge/Type-Feat-green)  
![Bug](https://img.shields.io/badge/Type-Bug-red) 

#### Jira ticket :blue_book:
https://cognitedata.atlassian.net/browse/REV-843

## Description :pencil:
Exposed `setIconsVisibility` and made a debug UI for it in Examples. However, this feature/function was not working. We were changing visibility during icon culling, so any changed in visibility will be overwritten in `IconCollection` by either `setIconClustersByLOD` or `computeProximityPoints` on next update.

This was fixed by distinguishing between culled and visible. An icon is only visible if it is **not culled AND set as visible**. If an icon is invisible, it will not be added during culling.

**Note**: We currently have a bug where the icon that is entered should be hidden during transition, but due to the culling scheme this is no longer working (maybe it never did). This is also fixed by this PR. We also have a super rare bug where the icon you entered is "covering the screen" and consumes input. This will probably also be fixed by being able to set the entered icon as invisible.

## How has this been tested? :mag:
On Windows in Chrome. I have tested the new "hide all" UI, but also tested culling in general by entering/leaving 360 images.

## Test instructions :information_source:
Load a set of 360 images:
- Hibernia_RS2
- helideck-site-2-jpeg
- helideck-site-2

1. Click icons to enter and use esc to leave. Make sure that the icon you left are still visible, and that culling looks correct in both states.
2. Use the "Hide all 360 Images" in the "360 Images" UI. Try this both outside and inside 360 images. Make sure that icons are only visible when they should be.


## Checklist :ballot_box_with_check:
